runtime/local/git/git.go

```diff
@@ -26,45 +28,102 @@ type binaryGitter struct {
+	// and probably is faster than downloading the whole repo history,
+	// but it comes with a bit of custom code for EACH host.
+	// @todo probably we should fall back to git in case the archives are not available.
+
 	if branchOrCommit == "latest" {
 		branchOrCommit = "master"
 	}
-	// @todo if it's a commit it must not be checked out all the time
-	repoFolder := strings.ReplaceAll(strings.ReplaceAll(repo, "/", "-"), "https://", "")
-	g.folder = filepath.Join(os.TempDir(),
-		repoFolder+"-"+shortid.MustGenerate())
-
-	url := fmt.Sprintf("%v/archive/%v.zip", repo, branchOrCommit)
-	if !strings.HasPrefix(url, "https://") {
-		url = "https://" + url
-	}
-	resp, err := http.Get(url)
-	if err != nil {
-		return fmt.Errorf("Can't get zip: %v", err)
-	}
+	if strings.Contains(repo, "github") {
+		// @todo if it's a commit it must not be checked out all the time
+		repoFolder := strings.ReplaceAll(strings.ReplaceAll(repo, "/", "-"), "https://", "")
+		g.folder = filepath.Join(os.TempDir(),
+			repoFolder+"-"+shortid.MustGenerate())
+
+		url := fmt.Sprintf("%v/archive/%v.zip", repo, branchOrCommit)
+		if !strings.HasPrefix(url, "https://") {
+			url = "https://" + url
+		}
+		resp, err := http.Get(url)
+		if err != nil {
+			return fmt.Errorf("Can't get zip: %v", err)
+		}
 
-	defer resp.Body.Close()
-	// Github returns 404 for tar.gz files...
-	// but still gives back a proper file so ignoring status code
-	// for now.
-	//if resp.StatusCode != 200 {
-	//	return errors.New("Status code was not 200")
-	//}
-
-	src := g.folder + ".zip"
-	// Create the file
-	out, err := os.Create(src)
-	if err != nil {
-		return fmt.Errorf("Can't create source file %v src: %v", src, err)
-	}
-	defer out.Close()
+		defer resp.Body.Close()
+		// Github returns 404 for tar.gz files...
+		// but still gives back a proper file so ignoring status code
+		// for now.
+		//if resp.StatusCode != 200 {
+		//	return errors.New("Status code was not 200")
+		//}
+
+		src := g.folder + ".zip"
+		// Create the file
+		out, err := os.Create(src)
+		if err != nil {
+			return fmt.Errorf("Can't create source file %v src: %v", src, err)
+		}
+		defer out.Close()
 
-	// Write the body to file
-	_, err = io.Copy(out, resp.Body)
-	if err != nil {
-		return err
+		// Write the body to file
+		_, err = io.Copy(out, resp.Body)
+		if err != nil {
+			return err
+		}
+		return unzip(src, g.folder, true)
+	} else if strings.Contains(repo, "gitlab") {
+		// Example: https://gitlab.com/micro-test/basic-micro-service/-/archive/master/basic-micro-service-master.tar.gz
+		// @todo if it's a commit it must not be checked out all the time
+		repoFolder := strings.ReplaceAll(strings.ReplaceAll(repo, "/", "-"), "https://", "")
+		g.folder = filepath.Join(os.TempDir(),
+			repoFolder+"-"+shortid.MustGenerate())
+
+		tarName := strings.ReplaceAll(strings.ReplaceAll(repo, "gitlab.com/", ""), "/", "-")
+		url := fmt.Sprintf("%v/-/archive/%v/%v.tar.gz", repo, branchOrCommit, tarName)
```

Comment:
I wish I could not duplicate much of the code above but given different urls, zip formats and folder structures etc. I think it's not worth the effort.